targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/spi_api.c

```diff
@@ -0,0 +1,346 @@
+    if (SPI_Return) {
+        obj->error = SPI_EVENT_ERROR;
+        return;
+    }
+}
+
+
+/** Write a byte out in master mode and receive a value
+ *
+ * @param[in] obj   The SPI peripheral to use for sending
+ * @param[in] value The value to send
+ * @return Returns the value received during send
+ */
+int spi_master_write(spi_t *obj, int value)
+{
+    ADI_SPI_TRANSCEIVER transceive;
+    uint8_t         TxBuf;
+    uint8_t         RxBuf;
+    ADI_SPI_HANDLE  SPI_Handle;
+    ADI_SPI_RESULT  SPI_Return = ADI_SPI_SUCCESS;
+
+    TxBuf = (uint8_t)value;
+
+    transceive.pReceiver        = &RxBuf;
+    transceive.ReceiverBytes    = 1;        /* link transceive data size to the remaining count */
+    transceive.nRxIncrement     = 1;        /* auto increment buffer */
+    transceive.pTransmitter     = &TxBuf;   /* initialize data attributes */
+    transceive.TransmitterBytes = 1;        /* link transceive data size to the remaining count */
+    transceive.nTxIncrement     = 1;        /* auto increment buffer */
+
+    transceive.bDMA = false;
+    transceive.bRD_CTL = false;
+    SPI_Handle = *obj->pSPI_Handle;
+    SPI_Return = adi_spi_MasterReadWrite(SPI_Handle, &transceive);
+    if (SPI_Return) {
+        obj->error = SPI_EVENT_ERROR;
+        return 1;
+    }
+
+    return((int)RxBuf);
+}
+
+
+/** Check if a value is available to read
+ *
+ * @param[in] obj The SPI peripheral to check
+ * @return non-zero if a value is available
+ */
+int spi_slave_receive(spi_t *obj)
+{
+    return 0;
+}
+
+
+/** Get a received value out of the SPI receive buffer in slave mode
+ *
+ * Blocks until a value is available
+ * @param[in] obj The SPI peripheral to read
+ * @return The value received
+ */
+int spi_slave_read(spi_t *obj)
+{
+    return 0;
+}
+
+
+/** Write a value to the SPI peripheral in slave mode
+ *
+ * Blocks until the SPI peripheral can be written to
+ * @param[in] obj   The SPI peripheral to write
+ * @param[in] value The value to write
+ */
+void spi_slave_write(spi_t *obj, int value)
+{
+    return;
+}
+
+
+/** Write a block out in master mode and receive a value
+ *
+ *  The total number of bytes sent and recieved will be the maximum of
+ *  tx_length and rx_length. The bytes written will be padded with the
+ *  value 0xff.
+ *
+ * @param[in] obj        The SPI peripheral to use for sending
+ * @param[in] tx_buffer  Pointer to the byte-array of data to write to the device
+ * @param[in] tx_length  Number of bytes to write, may be zero
+ * @param[in] rx_buffer  Pointer to the byte-array of data to read from the device
+ * @param[in] rx_length  Number of bytes to read, may be zero
+ * @param[in] write_fill Default data transmitted while performing a read
+ * @returns
+ *      The number of bytes written and read from the device. This is
+ *      maximum of tx_length and rx_length.
+ */
+int spi_master_block_write(spi_t *obj, const char *tx_buffer, int tx_length, char *rx_buffer, int rx_length, char write_fill)
+{
+    return 0;
```

Comment:
why is this not implemented yet?

Reply:
It was not needed yet.

Reply:
> It was not needed yet.

I do not follow. Please implement it, should be trivial.

Reply:
In the file 'hal/spi_api.h', the first line of the description for this function states "Write a block out in master mode and receive a value".  What is the purpose of the arguments "rx_length" and "write_fill" ?  Is this function intended to be used for a block read as well?